packages/wdio-allure-reporter/src/index.js

```diff
@@ -101,12 +101,23 @@ class AllureReporter extends WDIOReporter {
         const currentTest = this.allure.getCurrentTest()
 
         if (!this.isMultiremote) {
-            const { browserName, deviceName } = this.capabilities
-            const targetName = browserName || deviceName || cid
-            const browserstackVersion = this.capabilities.os_version || this.capabilities.osVersion
-            const version = browserstackVersion || this.capabilities.version || this.capabilities.platformVersion || ''
-            const paramName = deviceName ? 'device' : 'browser'
-            const paramValue = version ? `${targetName}-${version}` : targetName
+            const { browserName, device, deviceName, version, platformVersion } = this.capabilities
+
+            let targetName = browserName || device || deviceName || cid
+            let paramVersion = version || platformVersion || ''
+            let paramName = device || deviceName ? 'device' : 'browser'
+
+            if (this.config.hostname.includes('browserstack')) {
+                // eslint-disable-next-line camelcase
+                const { browser_name: browserNameLegacy, os_version: osVersionLegacy, osVersion: osVersionW3C } = this.capabilities
+                const { version: mobileVersion } = this.capabilities.mobile
+
+                targetName = browserNameLegacy || mobileVersion || device || deviceName || cid
+                paramVersion = osVersionW3C || osVersionLegacy || version || platformVersion || ''
+                paramName = mobileVersion || device || deviceName ? 'device' : 'browser'
```

Comment:
All these `||` conditionals scare me, this can make it very complicated to read and understand if anyone has to refactor / look into this before. Are you sure this is the best way to go about this?

Reply:
Not 100% sure I mean this is to allow for redundancy of these values in report, but I understand the extend of the `||` is a bit intense. I will take a look at this again.